modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/MeResourceManagerTest.java

```diff
@@ -0,0 +1,1553 @@
+
+        } else if (expectedScimResponseStatus == badRequest) {
+
+            doThrow(new BadRequestException()).when(userManager).deleteMe(userName);
+
+            when(AbstractResourceManager.encodeSCIMException(any(BadRequestException.class)))
+                    .thenReturn(getEncodeSCIMExceptionObject(new BadRequestException()));
+
+            SCIMResponse outputScimResponse = meResourceManager.delete(userName, userManager);
+            Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+        }
+
+    }
+
+    @DataProvider(name = "dataForTestDeleteFailsINCharonExceptionOnly")
+    public Object[][] dataToTestDeleteCharonExceptionOnly()
+            throws BadRequestException, CharonException, InternalErrorException {
+
+        User user = getNewUser();
+        String username = user.getUsername();
+        return new Object[][]{
+                {username, charon}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestDeleteFailsINCharonExceptionOnly")
+    public void testDeleteCharonExceptionOnly(String userName, int expectedScimResponseStatus)
+            throws NotFoundException, NotImplementedException, BadRequestException, CharonException {
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(endpoint);
+
+        doThrow(new CharonException()).when(userManager).deleteMe(userName);
+
+        when(AbstractResourceManager.encodeSCIMException(any(CharonException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new CharonException()));
+
+        SCIMResponse outputScimResponse = meResourceManager.delete(userName, userManager);
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+
+    }
+
+    @DataProvider(name = "dataForListWithGet")
+    public Object[][] dataToListWithGet() {
+
+        return new Object[][]{
+                {null, 1, 2, null, null, "PRIMARY", "emails", null},
+                {null, 1, 2, null, null, "PRIMARY", "userName", null},
+                {null, 1, 2, null, null, "PRIMARY", "id", null}
+        };
+    }
+
+    @Test(dataProvider = "dataForListWithGet")
+    public void testListWithGet(String filter, int startIndexInt, int countInt,
+                                String sortBy, String sortOrder, String domainName, String attributes,
+                                String excludeAttributes) {
+
+        SCIMResponse outputScimResponse = meResourceManager.listWithGET(userManager, filter, startIndexInt,
+                countInt, sortBy, sortOrder, domainName, attributes, excludeAttributes);
+
+        Assert.assertNull(outputScimResponse);
+    }
+
+    @DataProvider(name = "dataForTestListWithPost")
+    public Object[][] dataToTestListWithPost() {
+
+        return new Object[][]{
+                {null},
```

Comment:
why do we need a data provider simply to pass only null to the testListWithPOST method?

Reply:
Fixed in commit d8f6f1c18502d1174c97a568b559f353a6b945a6